bittide-instances/src/Bittide/Instances/Si539xSpi.hs

```diff
@@ -20,13 +29,81 @@ si5391Spi ::
   "" :::
     ( "readByte" ::: Signal Basic125 (Maybe Byte)
     , "BUSY" ::: Signal Basic125 Busy
-    , "STATE" ::: Signal Basic125 (ConfigState Basic125 432)
+    , "STATE" ::: Signal Basic125 (ConfigState Basic125 592)
     , "" :::
       ( "SCLK" ::: Signal Basic125 Bool
       , "MOSI" ::: Signal Basic125 Bit
       , "CSB" ::: Signal Basic125 Bool)
   )
 si5391Spi clk rst extOp miso = withClockResetEnable clk rst enableGen $
-  si539xSpi testConfig (SNat @50000) extOp miso
+  si539xSpi testConfigAll200 (SNat @50000) extOp miso
 
 makeTopEntity 'si5391Spi
+
+-- | An instance that combines the following components
+-- * 'Si539xSpi' core to configure the @Si5395J@ chip over SPI and enable further SPI communication
+-- * a `resettableXilinxElasticBuffer` to obtain the datacounts for `callisto`.
+-- * `callistoClockControl` to run the control algorithm that adjusts the clock frequency.
+-- * A `spiFrequencyController` to convert `callisto`s `SpeedChange`s into `RegisterOperation`s
+-- for the `Si539xSpi`.
+-- * A `stabilityChecker` to indicate when the clocks are considered to be synchronized.
+callistoSpi ::
+  "CLK_125MHZ"      ::: Clock Basic125 ->
+  "clkRecovered"    ::: Clock Internal ->
+  "clkControlled"   ::: Clock External ->
+  "reset125"        ::: Reset Basic125 ->
+  "locked"          ::: Signal External Bool ->
+  "MISO"            ::: Signal Basic125 Bit ->
+  "" :::
+    ( "BUSY"        ::: Signal Basic125 Busy
+    , "DONE"        ::: Signal Basic125 Bool
+    , "EBPASS"      ::: Signal External Bool
+    , "EBSTABLE"    ::: Signal External Bool
+    , "" :::
+      ( "SCLK"      ::: Signal Basic125 Bool
+      , "MOSI"      ::: Signal Basic125 Bit
+      , "CSB"       ::: Signal Basic125 Bool)
+    )
+callistoSpi clk125 clkRecovered clkControlled rst125 locked miso =
+  (spiBusy, configState .==. pure Finished, ebMode .==. pure Pass, isStable, spiOut)
+ where
+  -- Spi core
+  (_, spiBusy, configState, spiOut) = withClockResetEnable clk125 configReset enableGen
+    si539xSpi testConfig6_200_5_20 (SNat @(Nanoseconds 75)) freqOp miso
```

Comment:
Could you comment on the significance of 75 nanoseconds? I assume that's a SkyWorks documented value?

Reply:
It's the minimum period for the SPI clock :) I added a comment.